drivers/include/mlx90393.h

```diff
@@ -0,0 +1,403 @@
+ * 
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{.c}
+ * while (1)
+ * {
+ *     mlx90393_data_t mag_data;
+ *
+ *     if (mlx90393_read(&dev, &mag_data) != MLX90393_SUCCESS) {
+ *         ... // error handling
+ *     }
+ *     ...
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * ## Output data format
+ * 
+ * Function #mlx90393_read returns magnetic field data in micro tesla and temperature
+ * in deci celsius in parameter \p data. The range of the magnetic field value depends on:
+ * - the analog chain gain defined by mlx90393_params_t::gain
+ * - the resolution defined by mlx90393_params_t::resolution
+ * 
+ * ## Using Interrupts
+ * 
+ * The MLX90393 sensor supports the use of data-ready interrupts to signal when data is available. 
+ * To use interrupts, connect the INT pin of the MLX90393 to the designated interrupt pin on the MCU, 
+ * as configured in the parameters.
+ * When calling the function #mlx90393_read, it will block the calling thread until an interrupt 
+ * is triggered. Once an interrupt is triggered, the driver handles the interrupt with an internal 
+ * ISR and then proceeds with reading and converting data and finally returns from the #mlx90393_read function.
+ * 
+ * ## Power Saving
+ *
+ * The MLX90393 sensor can be shutdown into idle mode when no continuous measurements are 
+ * required using the function #mlx90393_stop_cont. The power consumption is then reduced to 
+ * max. 5 uA. To restart the MLX90393 in previous continuous measurement mode, the 
+ * #mlx90393_start_cont function can be used.
+ * In single measurement mode, once the measurement is completed, the sensor transitions into 
+ * idle mode, awaiting a new command from the master to initiate another acquisition.
+ * 
+ * @{
+ *
+ * @file
+ *
+ * @author      Michael Ristau <michael.ristau@fh-erfurt.de>
+ */
+
+#ifndef MLX90393_H
+#define MLX90393_H
+
+#include "periph/gpio.h"
+#if MODULE_MLX90393_SPI
+#include "periph/spi.h"
+#elif MODULE_MLX90393_I2C
+#include "periph/i2c.h"
+#endif
+
+/* Add header includes here */
+
+#ifdef __cplusplus
+extern "C" //{
+#endif
+
+/**
+ * @brief   Data container of the MLX90393 sensor
+ */
+typedef struct {
+    int32_t x_axis;                  /**< Magnometer data from x-axis */
+    int32_t y_axis;                  /**< Magnometer data from y_axis */
+    int32_t z_axis;                  /**< Magnometer data from z_axis */
+    int16_t temp;                    /**< Temperature */
+} mlx90393_data_t;
+
+/**
+ * @brief   Measurement mode of the MLX90393 sensor
+ */
+typedef enum {
+    MLX90393_MODE_BURST,                        /**< Burst mode */
+    MLX90393_MODE_SINGLE_MEASUREMENT,           /**< Single measurement mode */
+    MLX90393_MODE_WAKE_UP_ON_CHANGE_ABSOLUTE,   /**< Wake-up on change mode, compared to first measurement */
+    MLX90393_MODE_WAKE_UP_ON_CHANGE_RELATIVE,   /**< Wake-up on change mode, compared to previous measurement */
```

Comment:
```suggestion
    MLX90393_MODE_WOC_ABSOLUTE,   /**< Wake-up on change mode, compared to first measurement */
    MLX90393_MODE_WOC_RELATIVE,   /**< Wake-up on change mode, compared to previous measurement */
```
Es sollte hier die Abkürzung WOC genutzt werden, um kürzere Symboldefinitionen zu nutzen und die absolute Grenze für die Zeilenlänge von 100 Zeichen einzuhalten. WOC hatten Sie als Abkürzung eingeführt, zudem steht es im Kommentar nochmal drin.